react/ManualPrice.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { FunctionComponent, Fragment, useState } from 'react'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { FormattedPrice } from 'vtex.formatted-price'
+import { Button, ButtonPlain, InputCurrency, Modal, Tag } from 'vtex.styleguide'
+import { useRuntime } from 'vtex.render-runtime'
+
+import { useItemContext } from './ItemContext'
+import { getFormattedPrice } from './utils/price'
+
+const ManualPrice: FunctionComponent = () => {
+  const {
+    item,
+    itemIndex,
+    shouldAllowManualPrice,
+    onSetManualPrice,
+  } = useItemContext()
+  const [isModalOpen, setIsModalOpen] = useState(false)
+  const [manualPrice, setManualPrice] = useState(item.manualPrice)
+  const {
+    culture: { currency, locale },
+  } = useRuntime()
+  const intl = useIntl()
+  const priceChanged =
+    item.sellingPrice === item.manualPrice && item.sellingPrice !== item.price
+
+  const submitManualPrice = (evt: React.FormEvent<HTMLFormElement>) => {
+    evt.preventDefault()
+
+    if (manualPrice != null && !isNaN(manualPrice)) {
+      onSetManualPrice(manualPrice!, itemIndex)
+    }
+
+    setIsModalOpen(false)
+  }
+
+  const handleManualPriceChange = (
+    evt: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const newManualPrice = parseFloat(evt.target.value)
```

Comment:
this won't work if the store is on another locale different than english, right? e.g.

```js
console.log(parseFloat('123.45')) // 123.45
console.log(parseFloat('123,45')) // 123
```

Reply:
actually, the `InputCurrency` component already formats the value to a float before calling this `handleManualPriceChange`, so you don't need this conversion

Reply:
Idk if I get what you saying. 🤔  Actually, I'm testing it using the `pt-BR` locale.

Reply:
Oh, I just saw the second message.

Reply:
But if I don't `parseFloat` the `evt.target.value`, I receive the following error. 

![image](https://user-images.githubusercontent.com/12574426/94570362-2879c500-0245-11eb-9b73-3158717354d8.png)

Reply:
This happens because I use `newManualPrice` to multiply it by `100`.

Reply:
ohh, I think the TS typings are incorrect, for this component the `evt.target.value` is already a number (you can see that the `InputCurrency` converts it [here](https://github.com/vtex/styleguide/blob/master/react/components/InputCurrency/index.js#L37)), so you can safely ignore this error

Reply:
noice!

Reply:
are you going to change this?

Reply:
I tried but without the `parseFloat` I couldn't build the app. Do you know a way to ignore this?